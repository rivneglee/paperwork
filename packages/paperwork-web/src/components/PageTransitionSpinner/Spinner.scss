.pwapp-page-transition-spinner {
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  left: 0;
  z-index: 999;
  height: 100vh;
  width: 100%;
}
