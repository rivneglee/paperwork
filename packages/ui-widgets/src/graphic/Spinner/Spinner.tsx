import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

interface Props {
  title?: string;
  type?: 'box' | 'donut' | 'ellipsis';
  size?: 's' | 'm' | 'l';
}

const Box: FunctionComponent = () => (
  <div className="pw-spinner-waterbox__box">
    <span className="pw-spinner-waterbox__inner"></span>
  </div>
);

const Donut: FunctionComponent = () => (
  <div className="pw-spinner-donut__donut"></div>
);

const Ellipsis: FunctionComponent = () => (
  <div className="pw-spinner-ellipsis">
    <span className="pw-spinner-ellipsis__dot"></span>
    <span className="pw-spinner-ellipsis__dot"></span>
    <span className="pw-spinner-ellipsis__dot"></span>
  </div>
);

const Spinner: FunctionComponent<Props> = ({
  title = 'LOADING',
  type = 'box',
  size = 'm',
}) => (
  <div className="pw-spinner">
    <div className={classNames(
      'pw-spinner__animation',
      size && `pw-spinner__animation--size-${size}`,
    )}>
      { type === 'box' && <Box /> }
      { type === 'donut' && <Donut /> }
      { type === 'ellipsis' && <Ellipsis /> }
    </div>
    <div className="pw-spinner__text">{title}</div>
  </div>
);

export default Spinner;
