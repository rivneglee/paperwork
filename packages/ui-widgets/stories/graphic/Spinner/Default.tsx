import React from 'react';

import { Spinner, Card } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
    <Card header={<h3>Spinner types</h3>}>
      <Spinner type="box" title="BOX"/>
      <Spinner type="donut" title="DONUT"/>
      <Spinner type="ellipsis" title="ELLIPSIS"/>
    </Card>
    <Card header={<h3>Spinner sizes</h3>}>
      <Spinner size="s" title="Small"/>
      <Spinner size="m" title="Medium"/>
      <Spinner size="l" title="Large"/>
      <Spinner type="donut" size="s" title="Small"/>
      <Spinner type="donut" size="m" title="Medium"/>
      <Spinner type="donut" size="l" title="Large"/>
      <Spinner type="ellipsis" size="s" title="Small"/>
      <Spinner type="ellipsis" size="m" title="Medium"/>
      <Spinner type="ellipsis" size="l" title="Large"/>
    </Card>
  </div>
);
