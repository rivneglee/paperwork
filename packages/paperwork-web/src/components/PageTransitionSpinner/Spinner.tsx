import React from 'react';

import { Spinner } from '@paperwork/ui-widgets';
import './Spinner.scss';

export default () => (
  <div className="pwapp-page-transition-spinner">
    <Spinner type="box" title="LOADING"/>
  </div>
);
