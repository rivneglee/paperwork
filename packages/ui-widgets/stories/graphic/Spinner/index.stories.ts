import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Spinner', module)
  .add('default', Default);
