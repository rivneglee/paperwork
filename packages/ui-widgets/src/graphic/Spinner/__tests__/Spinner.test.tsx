import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Spinner from '../Spinner';

describe('Spinner', () => {
  it('should render default spinner', () => {
    const wrapper: ShallowWrapper = shallow(
      <Spinner />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render donut', () => {
    const wrapper: ShallowWrapper = shallow(
      <Spinner type="donut" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render ellipsis', () => {
    const wrapper: ShallowWrapper = shallow(
      <Spinner type="ellipsis" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render small size spinner', () => {
    const wrapper: ShallowWrapper = shallow(
      <Spinner size="s" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render medium size spinner', () => {
    const wrapper: ShallowWrapper = shallow(
      <Spinner size="m" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render large size spinner', () => {
    const wrapper: ShallowWrapper = shallow(
      <Spinner size="l" />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
