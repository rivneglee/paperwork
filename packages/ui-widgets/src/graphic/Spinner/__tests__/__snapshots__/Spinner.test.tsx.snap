// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render default spinner 1`] = `
<div
  className="pw-spinner"
>
  <div
    className="pw-spinner__animation pw-spinner__animation--size-m"
  >
    <Box />
  </div>
  <div
    className="pw-spinner__text"
  >
    LOADING
  </div>
</div>
`;

exports[`Spinner should render donut 1`] = `
<div
  className="pw-spinner"
>
  <div
    className="pw-spinner__animation pw-spinner__animation--size-m"
  >
    <Donut />
  </div>
  <div
    className="pw-spinner__text"
  >
    LOADING
  </div>
</div>
`;

exports[`Spinner should render ellipsis 1`] = `
<div
  className="pw-spinner"
>
  <div
    className="pw-spinner__animation pw-spinner__animation--size-m"
  >
    <Ellipsis />
  </div>
  <div
    className="pw-spinner__text"
  >
    LOADING
  </div>
</div>
`;

exports[`Spinner should render large size spinner 1`] = `
<div
  className="pw-spinner"
>
  <div
    className="pw-spinner__animation pw-spinner__animation--size-l"
  >
    <Box />
  </div>
  <div
    className="pw-spinner__text"
  >
    LOADING
  </div>
</div>
`;

exports[`Spinner should render medium size spinner 1`] = `
<div
  className="pw-spinner"
>
  <div
    className="pw-spinner__animation pw-spinner__animation--size-m"
  >
    <Box />
  </div>
  <div
    className="pw-spinner__text"
  >
    LOADING
  </div>
</div>
`;

exports[`Spinner should render small size spinner 1`] = `
<div
  className="pw-spinner"
>
  <div
    className="pw-spinner__animation pw-spinner__animation--size-s"
  >
    <Box />
  </div>
  <div
    className="pw-spinner__text"
  >
    LOADING
  </div>
</div>
`;
